### DISCUSSION TITLE
Can anyone share its caddy/nginx docker compose https setup?

### DISCUSSION THREAD

**Discussion by thiswillbeyourgithub** [open-webui/open-webui#4443] [Q&A]
Hi,

I've not succedded in enabling Caddy for https using internal/self signed certificate to access open-webui on my local wifi.

Can anyone who made it work share their config? I'm using docker compose and would prefer using caddy or nginx, in the same docker compose file.

Thanks!

**Comment by msorenss** [comment]
This is my docker config served from a batfile. It works. 

@echo off
setlocal enabledelayedexpansion

:: Check for administrative privileges
net session >nul 2>&1
if %errorlevel% neq 0 (
    echo This script requires administrative privileges.
    echo Please run this script as an administrator.
    pause
    exit /b 1
)

:: Check for NVIDIA drivers and CUDA support
where nvidia-smi >nul 2>&1
if %errorlevel% neq 0 (
    echo NVIDIA drivers are not installed or nvidia-smi is not in PATH.
    echo Please install NVIDIA drivers from https://www.nvidia.com/Download/index.aspx
    echo After installation, you may need to restart your computer.
    pause
    exit /b 1
)

:: Check CUDA version
for /f "tokens=3" %%i in ('nvidia-smi ^| findstr "CUDA Version"') do set CUDA_VERSION=%%i
echo CUDA Version: %CUDA_VERSION%

:: Check for Docker Desktop with NVIDIA integration
docker info | findstr "Runtimes: nvidia" >nul
if %errorlevel% neq 0 (
    echo NVIDIA integration for Docker Desktop is not detected.
    echo Please ensure Docker Desktop is installed with NVIDIA GPU support.
    echo Visit: https://docs.docker.com/desktop/windows/wsl/#gpu-support
    pause
    exit /b 1
)

:: Define paths and variables
set CADDY_CONF_DIR=C:\temp\caddy-docker
set DATA_DIR=C:\temp\data
set NETWORK_NAME=webui-network
set OLLAMA_DIR=C:\Temp\Ollama
set DDNS_DOMAIN=youradress.ddns.com

:: Set Ollama environment variables in Windows PATH
echo Setting Ollama environment variables...
setx OLLAMA_FLASH_ATTENTION 1
setx OLLAMA_HOST 0.0.0.0
setx OLLAMA_MAX_LOADED_MODELS 4
setx OLLAMA_NUM_PARALLEL 4
setx OLLAMA_MODELS %Ollama_DIR%

:: Refresh environment variables
call refreshenv

:: Create directories if they don't exist
if not exist "%CADDY_CONF_DIR%" mkdir "%CADDY_CONF_DIR%"
if not exist "%OLLAMA_DIR%" mkdir "%OLLAMA_DIR%"
if not exist "%DATA_DIR%" mkdir "%DATA_DIR%"

:: Add entry to hosts file
echo 127.0.0.1 test.local >> %WINDIR%\System32\drivers\etc\hosts

:: Create Docker network if it doesn't exist
docker network inspect %NETWORK_NAME% >nul 2>&1 || (
    echo Creating Docker network %NETWORK_NAME%
    docker network create %NETWORK_NAME%
)

:: Open Firewall Ports
echo Opening firewall ports 80, 443, 8080, and 11434 for access
netsh advfirewall firewall add rule name="Allow Port 80" dir=in action=allow protocol=TCP localport=80
netsh advfirewall firewall add rule name="Allow Port 443" dir=in action=allow protocol=TCP localport=443
netsh advfirewall firewall add rule name="Allow Port 8080" dir=in action=allow protocol=TCP localport=8080
netsh advfirewall firewall add rule name="Allow Port 11434" dir=in action=allow protocol=TCP localport=11434

:: Stop and remove existing containers
for %%c in (open-webui pipelines caddy-webui ollama) do (
    echo Stopping %%c
    docker stop %%c
    echo Removing %%c
    docker rm %%c
)

:: Pull and run Ollama with NVIDIA support and environment variables
echo Pulling and running Ollama with NVIDIA support...
docker pull ollama/ollama:latest
docker run -d --name ollama --network %NETWORK_NAME% ^
    --gpus all ^
    -v "%OLLAMA_DIR%:/root/.ollama" ^
    -p 11434:11434 ^
    -e OLLAMA_FLASH_ATTENTION=1 ^
    -e OLLAMA_HOST=0.0.0.0 ^
    -e OLLAMA_MAX_LOADED_MODELS=4 ^
    -e OLLAMA_NUM_PARALLEL=4 ^
    --restart always ^
    ollama/ollama

:: Pull and run Open WebUI with CUDA support
echo Pulling and running Open WebUI with CUDA support...
docker pull ghcr.io/open-webui/open-webui:main
docker run -d --name open-webui --network %NETWORK_NAME% ^
    --gpus all ^
    -p 8080:8080 ^
    -e OLLAMA_BASE_URL=http://ollama:11434 ^
    -v "%DATA_DIR%:/app/backend/data" ^
    --restart always ^
    ghcr.io/open-webui/open-webui:main

:: Pull and run Pipelines
echo Pulling and running Pipelines...
docker pull ghcr.io/open-webui/pipelines:main
docker run -d --name pipelines --network %NETWORK_NAME% ^
    -p 9099:9099 ^
    --add-host=host.docker.internal:host-gateway ^
    -v "%DATA_DIR%\pipelines:/app/pipelines" ^
    --restart always ^
    ghcr.io/open-webui/pipelines:main

:: Create Caddyfile
echo Creating Caddyfile for local and DDNS access
(
echo {
echo     # Global options
echo     email example@example.com
echo     acme_ca https://acme-v02.api.letsencrypt.org/directory
echo }
echo.
echo # Public access via DDNS
echo %DDNS_DOMAIN% {
echo     encode gzip
echo     reverse_proxy open-webui:8080
echo }
echo.
echo # Local network access
echo test.local {
echo     encode gzip
echo     reverse_proxy open-webui:8080
echo     tls internal
echo }
echo.
echo # HTTP catch-all that redirects to HTTPS
echo :80 {
echo     redir https://{host}{uri} permanent
echo }
) > %CADDY_CONF_DIR%\Caddyfile

:: Pull and run Caddy
echo Pulling and running Caddy...
docker pull caddy:latest
docker run -d --name caddy-webui --network %NETWORK_NAME% ^
    -p 80:80 -p 443:443 ^
    -v "%CADDY_CONF_DIR%\Caddyfile:/etc/caddy/Caddyfile" ^
    -v caddy_data:/data ^
    -v caddy_config:/config ^
    --restart always ^
    caddy:latest

:: Verify GPU access
echo Verifying GPU access...
docker exec open-webui nvidia-smi
docker exec ollama nvidia-smi

:: Test Ollama
echo Testing Ollama Llama3.1...
docker exec -it ollama ollama run llama3.1:8b-instruct-q5_K_M "Hello, how are you?"
docker exec -it ollama ollama run llama3.1:8b "Hello, how are you?"
echo Testing Ollama Phi 3 Vision...
docker exec -it ollama ollama run llava-phi3 "Hello, how are you?"

:: Verify Ollama environment variables
echo Verifying Ollama environment variables...
docker exec ollama env | findstr OLLAMA

echo Removing unused Docker images...
docker image prune -f

echo Setup complete. Open WebUI and Ollama are now running with GPU support.
echo Caddy is configured for both local and DDNS access.

echo Access Open WebUI locally at https://test.local
echo Access Open WebUI remotely at https://%DDNS_DOMAIN%
echo Ollama is accessible at http://localhost:11434
pause
endlocal
